Gems/ROS2/Code/Include/ROS2/Manipulator/ManipulatorControllerComponent.h

```diff
@@ -0,0 +1,91 @@
+        using FollowJointTrajectory = control_msgs::action::FollowJointTrajectory;
+        FollowJointTrajectoryActionServer();
+        // std::shared_ptr<GoalHandleFollowJointTrajectory> GetGoal();
+
+        // bool m_goalExecution{false};
+        rclcpp_action::Server<FollowJointTrajectory>::SharedPtr m_actionServer;
+        std::shared_ptr<GoalHandleFollowJointTrajectory> m_goalHandle;
+
+        GoalStatus m_goalStatus = GoalStatus::Pending;
+
+    protected:
+        // callbacks for action_server_
+        rclcpp_action::GoalResponse goal_received_callback(
+            const rclcpp_action::GoalUUID & uuid, 
+            std::shared_ptr<const FollowJointTrajectory::Goal> goal);
+        rclcpp_action::CancelResponse goal_cancelled_callback(
+            const std::shared_ptr<GoalHandleFollowJointTrajectory> goal_handle);
+        void goal_accepted_callback(
+            std::shared_ptr<GoalHandleFollowJointTrajectory> goal_handle);
+
+    };
+
+    //! Component responsible for controlling a robotic arm made up of hinge joints.
+    class ManipulatorControllerComponent
+        : public AZ::Component
+        , public AZ::TickBus::Handler
+    {
+    public:
+
+        AZ_COMPONENT(ManipulatorControllerComponent, "{3da9abfc-0028-4e3e-8d04-4e4440d2e319}", AZ::Component); // , ManipulatorRequestBus::Handler);
+
+        // AZ::Component interface implementation.
+        ManipulatorControllerComponent() = default;
+        ~ManipulatorControllerComponent() = default;
+        void Activate() override;
+        void Deactivate() override;
+        static void GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required);
+        static void Reflect(AZ::ReflectContext* context);
+        void OnTick(float deltaTime, AZ::ScriptTimePoint time) override;
+
+    private:
+        void InitializeMap();
+        void InitializePid();
+        void InitializeCurrentPosition();
+        void KeepStillPosition(const uint64_t & deltaTimeNs);
+        void ExecuteTrajectory(const uint64_t & deltaTimeNs);
+        float GetJointPosition(const AZ::Component* hingeComponent);
+        float ComputeFFJointVelocity(float &currentPosition, float &desiredPosition, const rclcpp::Duration & duration);
+        float ComputePIDJointVelocity(float & currentPosition, float & desiredPosition, const uint64_t & deltaTimeNs, int & jointIndex);
+        void SetJointVelocity(AZ::Component * hingeComponent, float & desiredVelocity);
+
+        FollowJointTrajectoryActionServer m_actionServerClass;
+        bool m_initialized{false};
+        bool m_initializedTrajectory{false};
+        bool m_pidBoolean{false};
+        bool m_KeepStillPositionInitialize{false};
+        AZStd::vector<VehicleDynamics::PidConfiguration> m_pidConfigurationVector;
+        AZStd::unordered_map<AZ::Name, AZ::EntityId> m_hierarchyMap;
+        AZStd::unordered_map<AZ::Name, float> m_jointKeepStillPosition;
```

Comment:
This does not seem to generalize well to joints with multiple degrees of freedom.
Also, I believe it would be great if these were encapsulated in components of relevant joint entities instead.

Reply:
@danielemorra98 @adamdbrw Has this point been addressed by now? Otherwise, is it already covered in https://github.com/o3de/sig-simulation/issues/54 or an issue report?

Reply:
The current status only focuses on 1DOF hinge joints. Widening the horizon and considering other joint types would be the next step to pace. 
I believe the high-level controller should be responsible to store the frozen position of every joint in the tree since these data belongs to the control logic.